### PR TITLE
refactor the Game.gamePlayerDice() function

### PR DESCRIPTION
**Pull request by cgolubi1**
* Related to #822 and #2047, a suggested replacement refactor for #1981
* Jenkins: http://jenkins.buttonweavers.com:8080/job/buttonmen-cgolubi1/461/ (if it passes)

This splits the monolith `Game.gamePlayerDice()` function into a number of distinct functions, which ideally has the effect of splitting things cleanly into functions which gather information and functions which lay out particular elements, and making operation consistent across active and captured dice.

The refactor *should* be a total noop --- the only difference is that it reorders class attributes in a couple of cases, but it should not add or remove attributes in any case, but only move code around.

### CODE REVIEW

**Comment by blackshadowshade** [comment]
Okay, QUnit tests pass on both Firefox and Chrome. Behaviour with Shadow Warriors vs Noeh looks good on both browsers for:
- normal dice
- dizzy dice
- warrior dice
- captured dice
- invalid selection of dice being reselected after attack failure
- non-active game mat
- keyboard handler

I'm happy to merge this.

**Comment by blackshadowshade** [comment]
So, as mentioned in IRC, I find the excessive use of brackets in 'if' statements distracting when I'm reading the code. I'll mark these for you as in-line comments.

File: src/ui/js/Game.js [hunk 1]
```diff
@@ -2688,8 +2688,286 @@ Game.gamePlayerStatus = function(player, reversed, game_active) {
...
+ * @param  string  player       Whose is the die? ('player' or 'opponent')
+ * @param  string  dieStatus    Status of the die ('active' or 'captured')
+ * @param  boolean isClickable  Is the die clickable?
+ * @return object               jQuery containing the die DIV
+ */
+Game.createGameMatDieDiv = function(die, player, dieStatus, isClickable) {
+  var divOpts = {
+    'class': 'die_img',
+  };
+  if ((dieStatus == 'active') && !(isClickable)) {
```
Comment (blackshadowshade) [inline src/ui/js/Game.js]: nit: !isClickable

File: src/ui/js/Game.js [hunk 2]
```diff
@@ -2688,8 +2688,286 @@ Game.gamePlayerStatus = function(player, reversed, game_active) {
...
+  if (dieIndex) {
+    divOptions.id = dieIndex;
+  }
+
+  // Set descriptions for active, captured, and unclickable dice
+  if (dieStatus == 'captured') {
+    divOptions.title = dieClickableInfo.reason;
+  } else {
+    divOptions.title = die.description;
+    if ((player_active) && !(dieClickableInfo.isClickable)) {
```
Comment (blackshadowshade) [inline src/ui/js/Game.js]: nit: brackets

File: src/ui/js/Game.js [hunk 3]
```diff
@@ -2701,148 +2979,37 @@ Game.gamePlayerDice = function(player, player_active) {
...
-      dieContainerDiv.append(dieBorderDiv);
-      dieContainerDiv.append(dieRecipeDiv);
-    }
+    die = Api.game[player].activeDieArray[i];
+    dieIndex = Game.dieIndexId(player, i);
+    dieClickableInfo = Game.dieClickableInfo(
+      die, player, 'active', player_active);
+    isSelected = (('dieSelectStatus' in Game.activity) &&
+      (dieIndex in Game.activity.dieSelectStatus) &&
+      (Game.activity.dieSelectStatus[dieIndex]));
```
Comment (blackshadowshade) [inline src/ui/js/Game.js]: nit: brackets for the last line

File: src/ui/js/Game.js [hunk 4]
```diff
@@ -2688,8 +2688,286 @@ Game.gamePlayerStatus = function(player, reversed, game_active) {
...
+ *
+ * @param  string  player       Whose is the die? ('player' or 'opponent')
+ * @param  boolean isSelected   Is the die currently selected?
+ * @return object               jQuery containing the border DIV
+ */
+Game.createGameMatBorderDiv = function(player, isSelected) {
+  var borderDivOpts = {
+    'class': 'die_border',
+  };
+  if (!(isSelected)) {
```
Comment (blackshadowshade) [inline src/ui/js/Game.js]: nit: brackets

**Comment by cgolubi1** [comment]
Jenkins for nits: http://jenkins.buttonweavers.com:8080/job/buttonmen-cgolubi1/464/ (assuming it passes)

**Comment by blackshadowshade** [comment]
Thanks, looks fine.

**Pull request merged**